financialaid/views_test.py

```diff
@@ -0,0 +1,90 @@
+"""
+Tests for financialaid view
+"""
+from django.core.urlresolvers import reverse
+from django.db.models.signals import post_save
+
+from factory.django import mute_signals
+from rest_framework.status import HTTP_201_CREATED, HTTP_400_BAD_REQUEST
+from rest_framework.test import APIClient
+
+from financialaid.api_test import FinancialAidBaseTestCase
+from financialaid.models import FinancialAid, FinancialAidStatus
+from profiles.factories import ProfileFactory
+
+
+class FinancialAidViewTests(FinancialAidBaseTestCase, APIClient):
+    """
+    Tests for financialaid views
+    """
+    @classmethod
+    def setUpTestData(cls):
+        super().setUpTestData()
+        with mute_signals(post_save):
+            cls.profile2 = ProfileFactory.create()
+        cls.url = reverse("financialaid_api")
+
+    def setUp(self):
+        super().setUp()
+        self.client.force_login(self.profile.user)
+        self.data = {
+            "original_currency": "USD",
+            "program_id": self.program.id,
+            "original_income": 80000
+        }
+
+    def test_income_validation(self):
+        """
+        Tests IncomeValidationView post endpoint
+        """
+        # Test that a FinancialAid object is created and is not auto-approved
+        assert FinancialAid.objects.count() == 0
+        resp = self.client.post(self.url, self.data, format='json')
+        assert resp.status_code == HTTP_201_CREATED
+        assert FinancialAid.objects.count() == 1
+        financial_aid = FinancialAid.objects.first()
+        assert financial_aid.tier_program == self.tiers["50k"]
+        assert financial_aid.status == FinancialAidStatus.PENDING_DOCS
+        # Test that a FinancialAid object is created and is auto-approved
+        self.data["original_income"] = 200000
+        resp = self.client.post(self.url, self.data, format='json')
+        assert resp.status_code == HTTP_201_CREATED
+        assert FinancialAid.objects.count() == 2
+        financial_aid = FinancialAid.objects.last()
```

Comment:
Good point - I've split these up into two separate test cases since the order_by seems a little messy.